plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java

```diff
@@ -148,6 +151,10 @@ public ResultSet getTables(Connection connection, Optional<String> schemaName, O
                     return Optional.of(varcharColumnMapping(createUnboundedVarcharType(), true));
                 }
                 return Optional.of(defaultVarcharColumnMapping(columnSize, true));
+            case Types.TIMESTAMP:
+                TimestampType timestampType = createTimestampType(3);
```

Comment:
What about `toWriteMapping`? Without it pushdown won't happen. This path is only used to convert the value sent in the ResultSet to the correct Trino type.

Reply:
I remember current commit works without `toWriteMapping`, anyway I will add tests to prove it. Thanks for reviewing!

Reply:
For how to do pushdown verification take a look at tests that use `isFullyPushedDown`.

Reply:
@hashhar I found the problem, the `timestampColumnMappingUsingSqlTimestampWithRounding` we used is disabled for pushing down: https://github.com/trinodb/trino/blob/c647c317405565ebd84599953dbf3cfca42affec/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java#L478

I am considering adding a `FULL_PUSHDOWN` version for this function, any ideas?

Reply:
If you want full pushdown then use the version without rounding - with rounding if you enable full pushdown it'll lead to incorrect results.

Reply:
I didn't find a function named `timestampColumnMappingUsingSqlTimestamp` -- https://github.com/trinodb/trino/pull/6893

Reply:
https://github.com/trinodb/trino/pull/8474/commits/54421315d9e44acb03bd0d383f22d5ad7b44cdac#r665278760 @hashhar Can't find  timestampColumnMappingUsingSqlTimestamp in StandardColumnMappings.java. Were you referring to timeColumnMappingUsingSqlTime instead of timestampColumnMappingUsingSqlTimestamp? Using timeColumnMappingUsingSqlTime resulted in the following error while running the query with filter on time column

`trino:druid> explain analyze select __time from orders where orders.__time >= date '1995-01-01';
Query 20210713_071149_00001_k9wgq failed: line 1:63: Cannot apply operator: time(3) <= date
explain analyze select __time from orders where orders.__time >= date '1995-01-01'`

Reply:
@jibinpt the function has been renamed to `timestampColumnMappingUsingSqlTimestampWithRounding`, but note it is a `DISABLE_PUSHDOWN` version, to make it work you need to change the `DISABLE_PUSHDOWN` to `FULL_PUSHDOWN`.

Reply:
@jerryleooo @hashhar isn't pushdown disabled in timestampColumnMappingUsingSqlTimestampWithRounding due to correctness reasons(as per the comments)? In that case, how exactly are we planning to pushdown timefilter in case of Druid?

Reply:
Continued at https://github.com/trinodb/trino/pull/8474#discussion_r670985264.